src/spotify_to_tidal/sync.py

```diff
@@ -342,6 +361,238 @@ def get_new_tidal_favorites() -> List[int]:
     else:
         print("No new tracks to add to Tidal favorites")
 
+async def sync_artists(spotify_session: spotipy.Spotify, tidal_session: tidalapi.Session, config: dict):
+    """Synchronize user-followed artists from Spotify to Tidal."""
+    print("Loading followed artists from Spotify")
+
+    async def get_all_followed_artists() -> List[dict]:
+        """Fetch all followed artists from Spotify."""
+        followed_artists = []
+        after = None
+
+        while True:
+            response = await repeat_on_request_error(
+                lambda: asyncio.to_thread(spotify_session.current_user_followed_artists, after=after)
+            )
+            artists = response['artists']['items']
+            followed_artists.extend(artists)
+
+            if not response['artists']['cursors'].get('after'):
+                break
+
+            after = response['artists']['cursors']['after']
+
+        return followed_artists
+
+    async def search_artist_on_tidal(artist_name: str):
+        """Search for an artist on Tidal by name."""
+        try:
+            search_results = tidal_session.search(artist_name, models=[tidalapi.artist.Artist])
+            return search_results.get('artists', [])
+        except Exception as e:
+            print(f"Error searching for artist '{artist_name}' on Tidal: {e}")
+            return []
+
+    async def get_top_albums_from_spotify(artist_id: str) -> List[dict]:
+        """Fetch the top 5 albums of a Spotify artist."""
+        try:
+            albums = spotify_session.artist_albums(artist_id, album_type="album,single", limit=5)['items']
+            return albums
+        except Exception as e:
```

Comment:
In general it's very unlikely that you should need to catch a general `Exception` in a function like this. If there's a specific exception that needs to be handled here then please catch that specific exception. If not then just remove the `try` block.

Reply:
I would expect a maximum of ONE instance of catching `except Exception as e` per PROJECT

Reply:
Yeah, that right. I should have thrown a more specific exception like `spotipy.exceptions.SpotifyException`. But I removed this function in my last commit.

Reply:
There are still 6 other instances of catching `Exception` though ;)

Reply:
I refactored the code to remove most of the try catch blocks. The one remaining in `search_album_on_tidal` is needed as the method `get_albums_by_barcode` is raising an ObjectNotFound exception when it cannot find an album with the given UPC.